### PR TITLE
RegionYolo operation specification refactoring.

### PR DESCRIPTION
**Pull request by jdanieck**
### Details:
 - Added *Types* section
 - Refactor input/output tensor description to refer types section.

### Tickets:
 - *56032*

### CODE REVIEW

File: docs/ops/detection/RegionYolo_1.md [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 **Short description**: *RegionYolo* computes the coordinates of regions with probability for each class.
 
-**Detailed description**: This operation is directly mapped to the original YOLO layer. [Reference](https://arxiv.org/pdf/1612.08242.pdf)
+**Detailed description**: This operation is directly mapped to the [original YOLO layer](https://arxiv.org/pdf/1612.08242.pdf).
```
Comment (lazarevevgeny) [inline docs/ops/detection/RegionYolo_1.md:9]: The detailed description should have the description of the algorithm. Also, there was an interesting feature of the RegionYolo implementation that works differently from the paper. I think that @mitruska did the analysis and can provide more information.

Comment (lazarevevgeny) [inline docs/ops/detection/RegionYolo_1.md:9]: I think that we have rule of thumb that the text in the square brackets should be equal to the title of the link. So in our case it should look like:

> ... directly mapped to the [YOLO9000: Better, Faster, Stronger](https://arxiv.org/pdf/1612.08242.pdf) paper.

File: docs/ops/detection/RegionYolo_1.md [hunk 2]
```diff
@@ -78,15 +78,18 @@
...
-*   **1**: `data` - 4D input tensor with floating point elements and shape `[N, C, H, W]`. Required.
+*   **1**: `data` - 4D tensor of type `T` and shape `[N, C, H, W]`. **Required.**
 
 **Outputs**:
 
-*   **1**: output tensor of rank 4 or less that codes detected regions. Refer to the original YOLO paper to decode the output as boxes. `anchors` should be used to decode real box coordinates. If `do_softmax` is set to 0, then the output shape is `[N, (classes + coords + 1)*len(mask), H, W]`. If `do_softmax` is set to 1, then output shape is partially flattened and defined in the following way:
+*   **1**: tensor of type `T` and rank 4 or less that codes detected regions. Refer to the original YOLO paper to decode the output as boxes. `anchors` should be used to decode real box coordinates. If `do_softmax` is set to 0, then the output shape is `[N, (classes + coords + 1)*len(mask), H, W]`. If `do_softmax` is set to 1, then output shape is partially flattened and defined in the following way:
 
     flat_dim = data.shape[axis] * data.shape[axis+1] * ... * data.shape[end_axis]
     output.shape = [data.shape[0], ..., data.shape[axis-1], flat_dim, data.shape[end_axis + 1], ...]
```
Comment (lazarevevgeny) [inline docs/ops/detection/RegionYolo_1.md:88]: Can we wrap this text with some style modifier?

File: docs/ops/detection/RegionYolo_1.md [hunk 3]
```diff
@@ -78,15 +78,18 @@
 
 **Inputs**:
 
-*   **1**: `data` - 4D input tensor with floating point elements and shape `[N, C, H, W]`. Required.
+*   **1**: `data` - 4D tensor of type `T` and shape `[N, C, H, W]`. **Required.**
 
 **Outputs**:
 
-*   **1**: output tensor of rank 4 or less that codes detected regions. Refer to the original YOLO paper to decode the output as boxes. `anchors` should be used to decode real box coordinates. If `do_softmax` is set to 0, then the output shape is `[N, (classes + coords + 1)*len(mask), H, W]`. If `do_softmax` is set to 1, then output shape is partially flattened and defined in the following way:
+*   **1**: tensor of type `T` and rank 4 or less that codes detected regions. Refer to the original YOLO paper to decode the output as boxes. `anchors` should be used to decode real box coordinates. If `do_softmax` is set to 0, then the output shape is `[N, (classes + coords + 1)*len(mask), H, W]`. If `do_softmax` is set to 1, then output shape is partially flattened and defined in the following way:
```
Comment (lazarevevgeny) [inline docs/ops/detection/RegionYolo_1.md:85]: add spaces around the `*` character

**Review by lazarevevgeny — CHANGES_REQUESTED** [review]

File: docs/ops/detection/RegionYolo_1.md (hunk 1 above)
Comment (jdanieck) [reply docs/ops/detection/RegionYolo_1.md]: Changed in this [commit](https://github.com/openvinotoolkit/openvino/pull/5926/commits/c0fd5d8d5baf158ea39d4f88df108dd61c530496).

File: docs/ops/detection/RegionYolo_1.md (hunk 2 above)
Comment (jdanieck) [reply docs/ops/detection/RegionYolo_1.md]: Changed in this [commit](https://github.com/openvinotoolkit/openvino/pull/5926/commits/c0fd5d8d5baf158ea39d4f88df108dd61c530496).

File: docs/ops/detection/RegionYolo_1.md (hunk 3 above)
Comment (jdanieck) [reply docs/ops/detection/RegionYolo_1.md]: Changed in this [commit](https://github.com/openvinotoolkit/openvino/pull/5926/commits/c0fd5d8d5baf158ea39d4f88df108dd61c530496).

File: docs/ops/detection/RegionYolo_1.md (hunk 1 above)
Comment (jdanieck) [reply docs/ops/detection/RegionYolo_1.md]: @lazarevevgeny I spoke with @mitruska and she worked on different operator *ReorgYolo* (which also only link to paper in detail description). This operator was reviewed and closed in this form long time ago and I wanted to only add simple fixes like *Types* section. Do you really expect of me to learn internals of this complex operation and write *Detailed description* section ? :)

**Review by lazarevevgeny — APPROVED** [review]

File: docs/ops/detection/RegionYolo_1.md (hunk 1 above)
Comment (lazarevevgeny) [reply docs/ops/detection/RegionYolo_1.md]: Merged

**Pull request merged**